cmd/proxy/actions/app_proxy.go

```diff
@@ -11,7 +11,7 @@ func addProxyRoutes(app *buffalo.App, storage storage.Backend) error {
 	app.GET("/{module:.+}/@v/{version}.info", cacheMissHandler(versionInfoHandler(storage), app.Worker))
 	app.GET("/{module:.+}/@v/{version}.mod", cacheMissHandler(versionModuleHandler(storage), app.Worker))
 	app.GET("/{module:.+}/@v/{version}.zip", cacheMissHandler(versionZipHandler(storage), app.Worker))
-	app.POST("/admin/upload/{module:[a-zA-Z./]+}/{version}", uploadHandler(storage))
```

Comment:
maybe removing this restriction should be enough. i like having module and version identifiers in a path, it identifies the resource. so I would prefer finding a way how to make this work and use req body as last resort.
does this make sense?

Reply:
why not letting marawan handle this as this is wonderful first PR

Reply:
Sure no problem 😊

Reply:
Thanks @michalpristas and @robjloranger 😄  -- For now, I made the regex just like the other ones. @robjloranger can you point me to a regex or a doc as to how Go module import paths should be matched? thanks!

Reply:
i think this should be fine. can you also add `@v/`?

Reply:
@michalpristas happy to. But would that potentially confuse people to think that this path is part of Vgo's Download Protocol?

Reply:
yeah probably. and as i think about it we will not want this in prod. so maybe we should add FCB to use this enpoint only in dev

Reply:
Would you like me to add that check as part of this PR? Is there a global variable to indicate whether runtime is dev or prod? 

Another random question: According to the wiki, Zeus can "host private modules". So I assumed this is the endpoint to which you'd do that. So wouldn't you want it in production as well? 

Thanks!

Reply:
regarding env check: https://github.com/gomods/athens/blob/58d42a1b571d60c40313c91da3f4fd468d024dd0/pkg/config/env/go.go#L11

about private modules. in v2 we will allow uploading of custom packages, right now it should be all discoverable. so private means it sits in private VCS (private gitlab instance).
so you host your code in gitlab instance deployed within companys VPN.
your private proxy  is deployed in same VPN
so it is all visible.
another way how to understand `private` is as a private repo in github visible to organization. in this case we need to provide a way how to make this package available through proxy. we don't really have this scenario thought through. 
@arschles correct me if i'm wrong

Reply:
ah, then maybe once the CLI is removed, this endpoint can be removed as well? For now, I put a check on top of the route. Let me know if that's what you were looking for. Thanks!

As for `private` github repos, the proxy can just use a GitHub Token to get access to private repos. So when a company deploys the proxy, they can just provide that token and the proxy knows how to use it.